src/operators.jl

```diff
@@ -1,70 +1,39 @@
 using ColorVectorSpace: AbstractGray, TransparentGray, TransparentRGB
 
-const dotops = VERSION < v"0.6.0-dev.1839"
-
-if isdefined(Base.Broadcast, :containertype)
-    # Specialize the low-level broadcasting machinery for ImageMeta.
-    # This make it work for all operators, rather than specializing
-    # each operator individually.
-    if VERSION > v"0.7-"
-        Base.BroadcastStyle(::Type{<:ImageMeta}) = Broadcast.ArrayStyle{ImageMeta}()
-        function Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{ImageMeta}}, ::Type{ElType}) where ElType
-            Mt = imagemeta(bc.args...)
-            M = Mt[1]
-            if length(Mt) > 1
-                for i = 2:length(Mt)
-                    Mt[i] == M || return ambigop(:Broadcast)
-                end
-            end
-            # Use the properties field of img to create the output
-            ImageMeta(similar(Array{ElType}, axes(bc)), M.properties)
-        end
-    else
-        using Base.Broadcast: _broadcast_eltype, broadcast_indices
-        Base.Broadcast._containertype(::Type{<:ImageMeta}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{ImageMeta}, ::Type{ImageMeta}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{Array}, ::Type{ImageMeta}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{ImageMeta}, ::Type{Array}) = ImageMeta
-        Base.Broadcast.promote_containertype(::Type{ImageMeta}, ct) = ImageMeta
-        Base.Broadcast.promote_containertype(ct, ::Type{ImageMeta}) = ImageMeta
-        Base.Broadcast.broadcast_indices(::Type{ImageMeta}, A) = indices(A)
-        function Base.Broadcast.broadcast_c(f, ::Type{ImageMeta}, As...)
-            T = _broadcast_eltype(f, As...)
-            shape = broadcast_indices(As...)
-            Mt = imagemeta(As...)
-            M = Mt[1]
-            if length(Mt) > 1
-                for i = 2:length(Mt)
-                    Mt[i] == M || return ambigop(Symbol(f))
-                end
-            end
-            broadcast!(f, shareproperties(M, similar(data(M), T, shape)), As...)
+# Specialize the low-level broadcasting machinery for ImageMeta.
+# This make it work for all operators, rather than specializing
+# each operator individually.
+Base.BroadcastStyle(::Type{<:ImageMeta}) = Broadcast.ArrayStyle{ImageMeta}()
```

Comment:
Oh, you are a god for doing this. ImageMeta was one of the packages that motivated me to overhaul broadcasting in Base (and the `ArrayAndChar` type in the docs was motivated by it), but I never got back here to finish the job.

This implementation seems spot-on.

Reply:
This is @daschw's work from an earlier commit. Github's diff